src/main/java/duke/execution/Parser.java

```diff
@@ -0,0 +1,150 @@
+package duke.execution;
+
+import duke.exceptions.DukeException;
+import duke.execution.commands.*;
+import duke.models.Task;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+
+import java.util.Date;
+
+public class Parser {
+
+    private static final SimpleDateFormat DATE_FORMAT = new SimpleDateFormat("dd/MM/yyyy HHmm");
+
+    /**
+     * Parses a line to return a Command.
+     * @param line Line to be parsed.
+     * @return Command after parsing.
+     */
+    public static Command parse(String line) throws DukeException {
+        assert line != null;
+        String[] split = line.split(" ");
+        String rest = getRestOfInput(split);
+        return parseForCommand(split[0], rest);
+    }
+
+    /**
+     * Checks whether a string can be formatted to be a Date.
+     * @param txt String to be checked.
+     * @return Boolean whether the string can be formatted to be a Date.
+     */
+    public static boolean isDate(String txt) {
```

Comment:
We suggest that you isolate this method into an utility class to reduce dependencies between classes.